rst_files/mccall_model.rst

```diff
@@ -354,48 +345,45 @@ Our initial guess :math:`v` is the value of accepting at every given wage
 
 .. code:: julia
 
-    function plot_value_function_seq(ax,
-                                     c=25,
-                                     β=0.99,
-                                     num_plots=6)
+    function plot_value_function_seq(plt,
+                                     c = 25,
+                                     β = 0.99,
+                                     num_plots = 6)
 
-        v = w_vals ./ (1 - β)
+        plt = deepcopy(plt)
+        v = collect(w_vals ./ (1 - β))
         v_next = similar(v)
 
         for i in 1:num_plots
-            ax[:plot](w_vals, v, label="iterate $i")
+            plot!(plt, w_vals, v, label = "iterate $i")
 
             # Update guess
             for (j, w) in enumerate(w_vals)
                 stop_val = w / (1 - β)
                 cont_val = c + β * sum(v .* p_vals)
                 v_next[j] = max(stop_val, cont_val)
             end
-            v[:] = v_next
+            v[:] .= v_next
         end
-        ax[:legend](loc="lower right")
-
+        return plt
     end
 
-    fig, ax = plt.subplots(figsize=(9, 6.5))
-    plot_value_function_seq(ax)
-    plt.show()
+    plot_value_function_seq(plt)
+
 
 Here's more serious iteration effort, that continues until measured deviation
 between successive iterates is below `tol`
 
 
-
 .. code-block:: julia
 
-    function compute_reservation_wage(c::Float64,
-                                      β::Float64;
-                                      max_iter::Int64=500,
-                                      tol::Float64=1e-6)
+    function compute_reservation_wage(c, β;
+                                      max_iter = 500,
+                                      tol = 1e-6)
 
     # == First compute the value function == #
 
-    v = w_vals ./ (1 - β)
+    v = collect(w_vals ./ (1 - β))
```

Comment:
Is there a reason we need to `collect` here?

Reply:
It uses an in-place operation afterwards and thus it needs to be a `Vector` and not a `UnitRange`.

Reply:
Makes sense.